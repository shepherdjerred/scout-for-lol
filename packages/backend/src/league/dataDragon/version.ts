import { z } from "zod";
import { first } from "remeda";

const versions = z
  .array(z.string())
  .parse(
    await (
      await fetch("https://ddragon.leagueoflegends.com/api/versions.json")
    ).json(),
  );

export const latestVersion = first(versions) as string;

if (latestVersion === undefined) {
  throw new Error("latest version is undefined");
}
