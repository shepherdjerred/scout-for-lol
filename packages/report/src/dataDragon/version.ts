import { z } from "zod";
import { first } from "remeda";

const versions = z
  .array(z.string())
  .parse(await (await fetch("https://ddragon.leagueoflegends.com/api/versions.json")).json());

const firstVersion = first(versions);

if (firstVersion === undefined) {
  throw new Error("latest version is undefined");
}

export const latestVersion = firstVersion;
