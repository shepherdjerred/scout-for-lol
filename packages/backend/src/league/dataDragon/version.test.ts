import { latestVersion } from "./version.ts";
import { assertSnapshot } from "std/testing/snapshot.ts";

Deno.test("should be able to get version", async (t) => {
  await assertSnapshot(t, latestVersion);
});
