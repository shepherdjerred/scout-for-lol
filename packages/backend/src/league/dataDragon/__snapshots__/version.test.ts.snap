export const snapshot = {};

snapshot[`should be able to get version 1`] = `"14.23.1"`;
